dll/shellext/zipfldr/zipfldr.cpp

```diff
@@ -94,6 +87,28 @@ GetDefaultUserSendTo(PWSTR pszPath)
                             SHGFP_TYPE_DEFAULT, pszPath);
 }
 
+UINT GetZipCodePage(BOOL bUnZip)
+{
+    WCHAR szValue[16];
+    DWORD dwType, cbValue = sizeof(szValue);
+    UINT nDefaultCodePage = (bUnZip ? CP_ACP : CP_UTF8);
+
+    LONG error = SHGetValueW(HKEY_CURRENT_USER, L"Software\\ReactOS",
+                             (bUnZip ? L"UnZipCodePage" : L"ZipCodePage"),
+                             &dwType, szValue, &cbValue);
```

Comment:
I think unzip code page is not needed, because it will be automatically detected by the UTF-8 flag.

Reply:
In default settings, it uses the `CP_ACP` codepage. `CP_ACP` is zero but the meaning of `CP_ACP` has many variation.

- The Japanese `CP_ACP` codepage and the Chinese `CP_ACP` codepage are different.
- The Japanese `CP_ACP` codepage and the Korean `CP_ACP` codepage are different.
- The Chinese `CP_ACP` codepage and the Russian `CP_ACP` codepage are different.

etc.

Reply:
if you want to extract from a Chinese `CP_ACP` codepage Zip file in the Japanese system, this setting is useful.

Reply:
Ah I see, this is to force the codepage on extraction when UTF-8 is not used, nice.